src/components/input-panel/spec-editor/index.js

```diff
@@ -29,6 +30,16 @@ const mapDispatchToProps = function(dispatch) {
 
     updateEditorString: (val) => {
       dispatch(EditorActions.updateEditorString(val));
+    },
+
+    updateVegaSpecInVegaLite: (spec) => {
+      dispatch(EditorActions.changeMode(MODES.Vega));
```

Comment:
If this just changes the mode, why not change `updateVegaSpec` to change the mode to Vega?